src/javascript/app/pages/user/account/change_password.js

```diff
@@ -1,45 +1,254 @@
+            name: 'Google',
+        },
+        'Facebook': {
+            icon: 'ic-linked-facebook',
+            name: 'Facebook',
+        },
+        'Apple': {
+            icon: 'ic-linked-apple',
+            name: 'Apple',
+        },
+    };
+
+    const shouldSetTradingPassword = () => {
+        const { status } = State.getResponse('get_account_status');
+        if (Array.isArray(status)) {
+            return status.includes('trading_password_required');
+        }
+        return false;
+    };
+
+    const hasSocialSignup = () => {
+        const { social_identity_provider, status } = State.getResponse('get_account_status');
+        if (Array.isArray(status)) {
+            social_signup_identifier = toTitleCase(social_identity_provider);
+            return status.includes('social_signup');
+        }
+        return false;
+    };
 
     const init = () => {
-        FormManager.init(form_id, [
+        $binary_password_container.setVisibility(1);
+
+        // Handle change binary password
+        FormManager.init(binary_form_id, [
             { selector: '#old_password', validations: ['req', ['length', { min: 6, max: 25 }]], clear_form_error_on_input: true },
             { selector: '#new_password', validations: ['req', 'password', ['not_equal', { to: '#old_password', name1: localize('Current password'), name2: localize('New password') }], 'compare_to_email'] },
-
             { request_field: 'change_password', value: 1 },
         ]);
         FormManager.handleSubmit({
-            form_selector       : form_id,
-            fnc_response_handler: handler,
+            form_selector       : binary_form_id,
+            fnc_response_handler: changePasswordHandler,
+        });
+
+        // Handle forgot binary password
+        $(forgot_binary_pw_btn_id).off('click').on('click', () => {
+            onSentEmail('binary_password');
+        });
+    };
+
+    const getDialogMessage = (event_type) => {
+        switch (event_type) {
+            case 'unlink':
+                return localize('Check your Google account email and click the link in the email to proceed.');
+            case 'trading_password':
+                return localize('Please click on the link in the email to reset your trading password.');
+            default:
+                return localize('Please click on the link in the email to reset your binary password.');
+        }
+    };
+
+    const onSentEmail = (event_type) => {
+        const req = {
+            verify_email: Client.get('email'),
+            type        : event_type === 'trading_password' ? 'trading_platform_password_reset' : 'reset_password',
+        };
+        BinarySocket.send(req).then(() => {
+            Dialog.alert({
+                id               : 'sent_email_dialog',
+                localized_message: getDialogMessage(event_type),
+                localized_title  : localize('We’ve sent you an email'),
+                ok_text          : localize('Okay'),
```

Comment:
I think there is a typo in the design, we don’t have `Okay` anywhere else both in binary-static and deriv.
If it is actually `OK` we don’t need to add `ok_text`
```suggestion
```
<img width="889" alt="Screen Shot 2021-04-19 at 12 55 23 PM" src="https://user-images.githubusercontent.com/70194182/115205519-e778cd80-a10e-11eb-9446-edc4b40311c7.png">

Reply:
Sure, will ask design to update it